### PR TITLE
Upgrade metrics dependence and middleware

### PR DESCRIPTION
**Pull request by ppaanngggg**
Use my fork of chi-prometheus which fix PatternMiddleware bug and upgrade chi to v5.
Use PatternMiddleware and make password optional.

### CODE REVIEW

**Comment by darh** [comment]
What would be the reason for not submitting a PR to `766b/chi-prometheus`? I would prefer if we keep that one for now. If there is no response from the repo author and this is not merged in a reasonable time (let's say 1mo after PR is submitted). 

Thoughts?

File: pkg/api/server/metrics.go [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
-	return chiprometheus.NewMiddleware(name)
+	return chiprometheus.NewPatternMiddleware(name)
 }
 
 func metricsMount(r chi.Router, username, password string) {
 	r.Route("/metrics", func(r chi.Router) {
-		r.Use(basicauth.New("Metrics", map[string][]string{
-			username: {password},
-		}))
+		if username != "" && password != "" {
```
Comment (darh) [inline pkg/api/server/metrics.go]: You said in the PR description that you are making password optional?
How would this work if you want to use username but password is empty?

Comment (ppaanngggg) [reply pkg/api/server/metrics.go]: I want to enable basicauth only when `username` and `password` set

**Comment by ppaanngggg** [comment]
> What would be the reason for not submitting a PR to `766b/chi-prometheus`? I would prefer if we keep that one for now. If there is no response from the repo author and this is not merged in a reasonable time (let's say 1mo after PR is submitted).
> 
> Thoughts?

Good idea, I create a PR to `chi-prometheus` first.

File: pkg/api/server/metrics.go (hunk 1 above)
Comment (darh) [reply pkg/api/server/metrics.go]: TBH I have no knowledge about Prometheus' auth constraints, what is required or possible. 
But your comment about making password optional and the code are incompatible :)

**Pull request closed without merging**